ui/hooks/identity/useProfileSyncing/accountSyncing.test.tsx

```diff
@@ -1,70 +1,181 @@
 import {
-  useAccountSyncingEffect,
+  useAccountSyncing,
   useDeleteAccountSyncingDataFromUserStorage,
+  useShouldDispatchAccountSyncing,
 } from './accountSyncing';
-import * as ProfileSyncModule from './profileSyncing';
 
-describe('useDeleteAccountSyncingDataFromUserStorage()', () => {
-  it('should dispatch account sync data deletion', async () => {
-    const mockDeleteAccountSyncAction = jest.spyOn(
-      actions,
-      'deleteAccountSyncingDataFromUserStorage',
-    );
+type ArrangeMocksMetamaskStateOverrides = {
+  isSignedIn?: boolean;
+  isProfileSyncingEnabled?: boolean;
+  isUnlocked?: boolean;
+  useExternalServices?: boolean;
+  completedOnboarding?: boolean;
+  isAccountSyncingReadyToBeDispatched?: boolean;
+};
 
-    const { result } = renderHookWithProviderTyped(
-      () => useDeleteAccountSyncingDataFromUserStorage(),
-      {},
-    );
+const initialMetamaskState: ArrangeMocksMetamaskStateOverrides = {
+  isSignedIn: false,
+  isProfileSyncingEnabled: false,
+  isUnlocked: true,
+  useExternalServices: true,
+  completedOnboarding: true,
+  isAccountSyncingReadyToBeDispatched: true,
+};
+
+const arrangeMockState = (
+  metamaskStateOverrides?: ArrangeMocksMetamaskStateOverrides,
+) => {
+  const state = {
+    metamask: {
+      ...initialMetamaskState,
+      ...metamaskStateOverrides,
+    },
+  };
+
+  return { state };
+};
 
-    await act(async () => {
-      await result.current.dispatchDeleteAccountData();
+describe('useShouldDispatchAccountSyncing()', () => {
+  const testCases = (() => {
+    const properties = [
+      'isSignedIn',
+      'isProfileSyncingEnabled',
+      'isUnlocked',
+      'useExternalServices',
+      'completedOnboarding',
+      'isAccountSyncingReadyToBeDispatched',
+    ] as const;
+    const baseState = {
+      isSignedIn: true,
+      isProfileSyncingEnabled: true,
+      isUnlocked: true,
+      useExternalServices: true,
+      completedOnboarding: true,
+      isAccountSyncingReadyToBeDispatched: true,
+    };
+
+    const failureStateCases: {
+      state: ArrangeMocksMetamaskStateOverrides;
+      failingField: string;
+    }[] = [];
+
+    // Generate test cases by toggling each property
+    properties.forEach((property) => {
+      const state = { ...baseState, [property]: false };
+      failureStateCases.push({ state, failingField: property });
     });
 
-    expect(mockDeleteAccountSyncAction).toHaveBeenCalled();
+    const successTestCase = { state: baseState };
+
+    return { successTestCase, failureStateCases };
+  })();
+
+  it('should return true if all conditions are met', () => {
+    const { state } = arrangeMockState(testCases.successTestCase.state);
+    const hook = renderHookWithProviderTyped(
+      () => useShouldDispatchAccountSyncing(),
+      state,
+      undefined,
+      MetamaskIdentityProvider,
+    );
+    expect(hook.result.current).toBe(true);
+  });
+
+  testCases.failureStateCases.forEach(({ state, failingField }) => {
```

Comment:
Love the test cleanup!! 😍 

Instead of a `forEach` how about using `it.each()`?